operator/internal/controller/vectorized/cluster_controller_test.go

```diff
@@ -0,0 +1,414 @@
+			require.NoError(t, err)
+
+			if l, ok := obj.(client.ObjectList); ok {
+				lists = append(lists, l)
+			}
+		}
+	}
+
+	// Collect all resources that belong to our cluster
+	clusterSelector := labels.ForCluster(cluster).AsClientSelector()
+	for _, l := range lists {
+		err := ctl.List(ctx, l)
+		if errors.Is(err, &meta.NoKindMatchError{}) {
+			continue
+		}
+		require.NoError(t, err)
+
+		objs, err := meta.ExtractList(l)
+		require.NoError(t, err)
+
+		for _, obj := range objs {
+			o := obj.(client.Object)
+
+			if !clusterSelector.Matches(k8slabels.Set(o.GetLabels())) {
+				continue
+			}
+
+			allClusterResources = append(allClusterResources, o)
+			t.Logf("%T: %s/%s", o, o.GetNamespace(), o.GetName())
+		}
+	}
+
+	// Additionally collect specific resources by name that don't have cluster labels
+	// ServiceAccount
+	var sa corev1.ServiceAccount
+	err = ctl.Get(ctx, client.ObjectKey{Name: "cluster", Namespace: "default"}, &sa)
+	if err == nil {
+		allClusterResources = append(allClusterResources, &sa)
+		t.Logf("%T: %s/%s", &sa, sa.GetNamespace(), sa.GetName())
+	}
+
+	// ClusterRole
+	var cr rbacv1.ClusterRole
+	err = ctl.Get(ctx, client.ObjectKey{Name: "redpanda-init-configurator"}, &cr)
+	if err == nil {
+		allClusterResources = append(allClusterResources, &cr)
+		t.Logf("%T: %s/%s", &cr, cr.GetNamespace(), cr.GetName())
+	}
+
+	// ClusterRoleBinding
+	var crb rbacv1.ClusterRoleBinding
+	err = ctl.Get(ctx, client.ObjectKey{Name: "redpanda-init-configurator"}, &crb)
+	if err == nil {
+		allClusterResources = append(allClusterResources, &crb)
+		t.Logf("%T: %s/%s", &crb, crb.GetNamespace(), crb.GetName())
+	}
+
+	// Clean up metadata fields that vary between test runs
+	for _, resource := range allClusterResources {
+		// Clear volatile metadata fields
+		resource.SetResourceVersion("")
+		resource.SetUID("")
+		resource.SetCreationTimestamp(metav1.Time{})
+		resource.SetManagedFields(nil)
+		resource.SetOwnerReferences(nil)
+
+		// Clear annotations that may contain volatile data
+		if annotations := resource.GetAnnotations(); annotations != nil {
+			delete(annotations, "redpanda.com/last-applied")
+			if len(annotations) == 0 {
+				resource.SetAnnotations(nil)
+			}
+		}
+
+		// Clear volatile fields specific to Services
+		if svc, ok := resource.(*corev1.Service); ok {
+			for i := range svc.Spec.Ports {
+				if svc.Spec.Ports[i].NodePort != 0 {
+					svc.Spec.Ports[i].NodePort = 30000 // Use consistent port for golden file
+				}
+			}
+			// Normalize ClusterIP addresses (except for headless services)
+			if svc.Spec.ClusterIP != "" && svc.Spec.ClusterIP != "None" {
+				svc.Spec.ClusterIP = "10.0.0.1" // Use consistent ClusterIP for golden file
+			}
+			// Normalize ClusterIPs array
+			for i := range svc.Spec.ClusterIPs {
+				if svc.Spec.ClusterIPs[i] != "None" {
+					svc.Spec.ClusterIPs[i] = "10.0.0.1" // Use consistent ClusterIP for golden file
+				}
+			}
+		}
+
+		if cm, ok := resource.(*corev1.ConfigMap); ok {
```

Comment:
This file _should_ be deterministic....  Do we need to await some other condition or update the generation to make it so?

If we can't do so immediately, I'd prefer that we omit the `Data` entirely and include a _very detailed_ note as to why we can't store the contents in the golden file.